### PR TITLE
Refactor to render SVG element with React

### LINKED ISSUE #183: Reduce code duplication between treemap and bubblemap

### LINKED ISSUE #212: Blob labels aren't bold when matched in search results
image

### LINKED ISSUE #214: Experiment with more performant rendering

### PR DESCRIPTION
**Pull request by joglr**
Resolves #214 
Resolves #212 
Resolves #183

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -14,14 +15,15 @@
     "@types/react": "^17.0.39",
     "@types/react-dom": "^17.0.11",
     "@types/styled-components": "^5.1.23",
+    "clsx": "^1.1.1",
```
Comment (joglr) [inline app/package.json:18]: This dependency is unused

File: app/src/components/Chart.css [hunk 2]
```diff
@@ -21,7 +32,6 @@
 }
 
 .search-match-title {
```
Comment (joglr) [inline app/src/components/Chart.css:34]: This declaration is unused

File: app/src/components/Chart.css [hunk 3]
```diff
@@ -38,6 +48,6 @@
   align-items: center;
 }
 
-svg {
+.name-text {
```
Comment (joglr) [inline app/src/components/Chart.css:51]: This declaration is unused

File: app/src/components/SearchBar.tsx [hunk 4]
```diff
@@ -12,21 +13,28 @@ export default function SearchBar(props: {
...
 
   useDebounce(
     () => {
-      props.setSearchText(value)
+      startTransition(() => {
+        props.setSearchText(value)
+      })
     },
-    100,
+    1,
```
Comment (joglr) [inline app/src/components/SearchBar.tsx:25]: Revert this

File: app/src/components/SidePanel.tsx [hunk 5]
```diff
@@ -8,7 +8,9 @@ import { Options } from "./Options"
 import SearchBar from "./SearchBar"
 import { Spacer } from "./Spacer"
 
-const SidePanelRoot = styled.aside``
+const SidePanelRoot = styled.aside`
+  overflow-y: auto;
+`
```
Comment (joglr) [inline app/src/components/SidePanel.tsx:13]: This introduces an issue when the sidebar is overflowing, a horizontal scrollbar is shown

File: app/src/util.ts [hunk 6]
```diff
@@ -16,6 +17,9 @@ export function unionAuthors(blob: HydratedGitBlobObject) {
   )
 }
 
-export function classes(...args: (string | undefined)[]) {
-  return args.filter((x) => x).join(" ")
+export function diagonal(d: HierarchyRectangularNode<unknown>) {
```
Comment (joglr) [inline app/src/util.ts:20]: This is unused

**Review by tjomson — COMMENTED** [review]
Search animations are different between bubble and tree map, is this intended?

When changing from tree to bubble map, there is a short delay before the folder names are shown.

Search field seems inconsistent. Sometimes very quick and smooth, other times laggy

**Review by hojelse — CHANGES_REQUESTED** [review]
Can we improve this?

1x click of a node or 1x character typed in search triggers 278 calls to unionAuthors()


- Providers() - 1x
  - getDefaultStore() - 1x
    - setupMetricsCache() foreach "blob" (72 blobs) - 72x
      - setDominanceColor() - 72x
      - setDominantAuthorColor() - 72x

both setDominanceColor and SetDominantAuthorColor calls unionAuthors() 

1 reload triggers 2x getDefaultStore() - 2 * (72 + 72) ≃ 278

1 click is sometimes a double click, triggering 4x getDefaultStore() - 4 * (72 + 72) = 576

**Comment by tjomson** [comment]
> Can we improve this?
> 
> 
> 
> 1 click of a svg node triggers:
> 
>     * Providers() - 1x
>       
>       * getDefaultStore() - 1x
>         
>         * setupMetricsCache() foreach "blob" (72 blobs) - 72x
>           
>           * setDominanceColor() - 72x
>             
>             * unionAuthors() - 72x
>           * setDominantAuthorColor() - 72x
>             
>             * unionAuthors() - 72x
> 
> 
> both setDominanceColor and SetDominantAuthorColor calls unionAuthors()
> 
> 1 browser reload triggers 2x Providers() calls - 2 * (72 + 72) ≃ 278
> 
> 1 click is sometimes a double click, triggering 4x Providers() calls - 4 * (72 + 72) = 576

**Comment by joglr** [comment]
> Search animations are different between bubble and tree map, is this intended?

I have made this more consistent.

> When changing from tree to bubble map, there is a short delay before the folder names are shown.

Issue was created for looking into this issue.

> Search field seems inconsistent. Sometimes very quick and smooth, other times laggy

I have readded the debounce.

**Review by tjomson — APPROVED** [review]
🚀🥌

**Pull request merged**